src/Bridge/Doctrine/Persister/ObjectManagerPersister.php

```diff
@@ -123,19 +116,23 @@ private function getPersistableClasses(ObjectManager $manager): array
         return $persistableClasses;
     }
 
-    protected function configureIdGenerator(ORMClassMetadataInfo $metadata): void
+    private function saveMetadataToRestore(ClassMetadata $metadata): void
     {
-        $metadata->setIdGeneratorType(ORMClassMetadataInfo::GENERATOR_TYPE_NONE);
-        $metadata->setIdGenerator(new ORMAssignedGenerator());
+        if (!isset($this->metadataToRestore[$metadata->getName()])) {
+            $this->metadataToRestore[$metadata->getName()] = $metadata;
+        }
     }
 
-    private function getMetadata(string $class): ClassMetadata
+    private function configureIdGenerator(ORMClassMetadataInfo $metadata): ORMClassMetadataInfo
```

Comment:
The `configureIdGenerator` has no reason to be `protected` AFAIK, so I make it private and changed its signature.